src/app/components/user-profile/user-profile.component.ts

```diff
@@ -17,40 +25,39 @@ export class UserProfileComponent implements OnInit {
   userProfileForm: FormGroup;
   countries: MyOption<CountriesEnum, string>[];
   provinces: MyOption<ProvincesEnum, string>[] = [];
-  formValues: FormValues = {name: '', isOld: '', isYoung: '', phoneNumber: '', country: '', province: ''};
+  doYouLikeTrump: MyOption<TrumpEnum, string>[];
+  formValues: FormValues = {name: '', isOld: '', isYoung: '', phoneNumber: '', country: '', province: '', trump: ''};
   showFormValues = false;
 
   constructor(private formBuilder: FormBuilder, private translateService: TranslateService) {
   }
 
   ngOnInit(): void {
+    this.userProfileForm = this.createFormGroup();
     const getCanada = mapCountryTranslate.get(CountriesEnum.CANADA);
     const getUS = mapCountryTranslate.get(CountriesEnum.US);
+    const getTrumpYes = mapTrumpTranslate.get(TrumpEnum.YES);
+    const getTrumpNo = mapTrumpTranslate.get(TrumpEnum.NO);
 
-
+// Load Country Dropdown and Trump Radio btn depending on language selected
     this.translateService.onLangChange.subscribe((event: LangChangeEvent) => {
-      console.log('getCanada', getCanada);
-
       this.translateService.get([getCanada, getUS])
         .subscribe((countriesTranslation) => {
-          console.log('countriesTranslation', countriesTranslation);
-          console.log('countriesTranslation', countriesTranslation.canada);
           this.countries = [{label: countriesTranslation[getCanada], value: CountriesEnum.CANADA},
             {label: countriesTranslation[getUS], value: CountriesEnum.US}];
         });
+      this.translateService.get([getTrumpYes, getTrumpNo])
+        .subscribe(trumpTranslation => {
+          this.doYouLikeTrump = [{label: trumpTranslation[getTrumpYes], value: TrumpEnum.YES},
+            {label: trumpTranslation[getTrumpNo], value: TrumpEnum.NO}];
+        });
     });
 
-
-    this.userProfileForm = this.createFormGroup();
+    // Load Dropdown Options on Country Change
     this.userProfileForm.get('dropdownGroup.country').valueChanges
       .subscribe(country => {
         if (country) {
           const provinces = mapProvinces.get(country);
-          const translations$ = provinces.map(key => {
-            return this.translateService.get(mapProvinceTranslate.get(key));
-          });
-          combineLatest(translations$).subscribe(translation => {
-          });
           this.provinces = provinces.map((province, index) => {
             return {value: province, label: provinces[index]};
```

Comment:
Well ... Does `provinces[index]` hold the actual translations ?

Reply:
You're right